microsoft/testsuites/cpu/cpusuite.py

```diff
@@ -51,3 +58,87 @@ class CPUSuite(TestSuite):
     )
     def verify_cpu_hot_plug(self, log: Logger, node: Node) -> None:
         verify_cpu_hot_plug(log, node)
+
+    @TestCaseMetadata(
+        description="""
+            This test will check cpu hotplug with storage workload.
+            The cpu hotplug steps are same as `verify_cpu_hot_plug` test case.
+            """,
+        priority=3,
+    )
+    def verify_cpu_offline_storage_workload(self, log: Logger, node: Node) -> None:
+        # run fio process asynchronously on the node
+        try:
+            fio_process = node.tools[Fio].launch_async(
+                name="workload",
+                filename="fiodata",
+                mode="readwrite",
+                iodepth=128,
+                numjob=10,
+                time=300,
+                block_size="1M",
+                size_gb=1,
+                group_reporting=False,
+                overwrite=True,
+                time_based=True,
+            )
+
+            # FIO output emits lines of the following form when it is running
+            # (f=10): [M(10)][24.9%][r=70.0MiB/s,w=75.0MiB/s]
+            # [r=70,w=75 IOPS][eta 03m:46s]
+            # check if stdout buffers contain the string "eta" to
+            # determine if it is running
+            fio_process.wait_warmup("eta")
+
+            # verify cpu hotplug functionality
+            verify_cpu_hot_plug(log, node)
+
+            # verify that the fio was running when hotplug was triggered
+            assert_that(
+                fio_process.is_running(),
+                "Storage workload was not running during CPUhotplug",
+            ).is_true()
+        finally:
+            # kill fio process
+            node.tools[Kill].by_name("fio")
+
+    @TestCaseMetadata(
+        description="""
+            This test will check cpu hotplug with network workload.
+            The cpu hotplug steps are same as `verify_cpu_hot_plug` test case.
+            """,
+        priority=3,
+        requirement=simple_requirement(
+            min_count=2,
+        ),
+    )
+    def verify_cpu_offline_network_workload(
+        self, log: Logger, environment: Environment
+    ) -> None:
+        server = cast(RemoteNode, environment.nodes[0])
+        client = cast(RemoteNode, environment.nodes[1])
+
+        try:
+            # run iperf3 process asynchronously on the server and client
+            server.tools[Iperf3].run_as_server_async()
+            client_iperf_process = client.tools[Iperf3].run_as_client_async(
+                server.internal_address, run_time_seconds=300
```

Comment:
it doesn't need to set timeout here. Keep it running.

Reply:
Sorry I didn't get this. The run_time here is the total time for iperf to run. Do you mean I should set this to infinity?

Reply:
Yes, set it to infinity. So if the following logic takes place more than 300 seconds, the test logic is still validated. There is an overall timeout on test case level, it will prevent the test case keeps running.